### DISCUSSION TITLE
Using the include API when eagerly fetching nullable relations

### DISCUSSION THREAD

**Discussion by adcooley** [prisma/prisma#6540] [Q&A] [answered]
We have models defined with optional relations, eg:

```prisma
model Author {
  id
  publisherId    String?             @db.Uuid
  publisher       Publisher?       @relation(fields: [publisherId], references: [id])
}
```

then when requesting the author, we `include` the publisher

```ts
const author = await prisma.author.findUnique({
  where: { id: "my-id" },
  include: {
    pubilsher: true,
  }
});
```

The reported type of `author` is 

```ts
Author & {
  publisher: Publisher
}
```

Why isn't the publisher relation optional in the generated type? This will quickly lead to runtime issues where our code expects a non-optional publisher but the database may return null. Shouldn't it generate the following type?

```ts
Author & {
  publisher: Publisher | null
}
```

**Comment by ryands17** [comment]
@adcooley 👋 

I tried the above and I'm getting the correct typing.

![image](https://user-images.githubusercontent.com/19697099/114518067-983d2380-9c5c-11eb-88b9-bbce3c9eecf5.png)

Could you share your `tsconfig.json` and you should have either `strict` or `strictNullChecks` set to `true`.

**Reply by adcooley** [reply, marked as answer]
amazing - thank you so much! I forgot to add strictNullChecks in my new project…you're a savior 🙏

**Reply by adcooley** [reply]
Follow up question for you - if I change the schema back now to a required relation

```prisma
model Author {
  id
  publisherId    String              @db.Uuid
  publisher       Publisher       @relation(fields: [publisherId], references: [id])
}
```

I'm getting a non-optional `publisher` when using the include api 👍 

```ts
const { publisher } = await context.prisma.author.findUnique({
  where: { id: parent.id },
  include: { publisher: true },
  rejectOnNotFound: true,
});
return customer;
```

However, the fluent API (which is a much cleaner syntax imo) returns a non-optional

```ts
return context.prisma.author.findUnique({
  where: { id: parent.id },
  rejectOnNotFound: true,
}).publisher()
```

The type of the returned publisher is `Publisher | null`

**Reply by ryands17** [reply]
It's a known issue :)
https://github.com/prisma/prisma/issues/5715

**Reply by adcooley** [reply]
woot ty subscribed :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
